x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/rule_definition_section.tsx

```diff
@@ -181,17 +305,60 @@ const prepareDefinitionSectionListItems = (
     });
   }
 
+  if (savedQuery) {
+    definitionSectionListItems.push({
+      title: descriptionStepI18n.SAVED_QUERY_NAME_LABEL,
+      description: <SavedQueryName savedQueryName={savedQuery.attributes.title} />,
+    });
+
+    if (savedQuery.attributes.filters) {
+      definitionSectionListItems.push({
+        title: descriptionStepI18n.SAVED_QUERY_FILTERS_LABEL,
+        description: <Filters filters={savedQuery.attributes.filters as Filter[]} />,
+      });
+    }
+  }
+
+  if ('filters' in rule && rule.filters && rule.filters.length > 0) {
+    definitionSectionListItems.push({
+      title: savedQuery
+        ? descriptionStepI18n.SAVED_QUERY_FILTERS_LABEL
+        : descriptionStepI18n.FILTERS_LABEL,
+      description: (
+        <Filters
+          filters={rule.filters as Filter[]}
```

Comment:
Is the `as` casting here required because the rule is `Partial<RuleResponse>`?

Reply:
It's because `rule.filters` are typed as an array of unknowns in `RuleResponse` schema [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.ts)
```typescript
/**
 * TODO: Right now the filters is an "unknown", when it could more than likely
 * become the actual ESFilter as a type.
 */
export type RuleFilterArray = t.TypeOf<typeof RuleFilterArray>; // Filters are not easily type-able yet
export const RuleFilterArray = t.array(t.unknown); // Filters are not easily type-able yet
```

And when using it for the Rule Details page it's cast as `Filter[]` [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/helpers.tsx)
```typescript
...
  queryBar: {
    query: { query: rule.query ?? '', language: rule.language ?? '' },
    filters: (rule.filters ?? []) as Filter[],
    saved_id: rule.saved_id ?? null,
  },
...
```
So I just borrowed the approach from the Rule Details page. It's definitely imperfect. Do you think we can improve it?

Reply:
Ah okay - I think this is fine then 👍